fuzz_lightyear/datastore.py

```diff
@@ -47,23 +55,36 @@ def register_post_fuzz_hook(
         tags = []
 
     for operation_id in operation_ids:
-        _POST_FUZZ_HOOKS_BY_OPERATION[operation_id].add(hook)
+        _ALL_POST_FUZZ_HOOKS_BY_OPERATION[operation_id].add(hook)
+        if rerun:
+            _RERUN_POST_FUZZ_HOOKS_BY_OPERATION[operation_id].add(hook)
 
     for tag in tags:
-        _POST_FUZZ_HOOKS_BY_TAG[tag].add(hook)
+        _ALL_POST_FUZZ_HOOKS_BY_TAG[tag].add(hook)
+        if rerun:
+            _RERUN_POST_FUZZ_HOOKS_BY_TAG[tag].add(hook)
 
 
 def get_post_fuzz_hooks(
     operation_id: str,
     tag: Optional[str] = None,
+    rerun: Optional[bool] = False,
 ) -> List[PostFuzzHook]:
     """Returns a list of functions that should be applied to fuzzed
     data for the input operation.
     """
-    operation_hooks = _POST_FUZZ_HOOKS_BY_OPERATION[operation_id].union(
-        _POST_FUZZ_HOOKS_BY_OPERATION['*'],
+    operation_hook_store = _RERUN_POST_FUZZ_HOOKS_BY_OPERATION \
+        if rerun \
+        else _ALL_POST_FUZZ_HOOKS_BY_OPERATION
+
+    tag_hook_store = _RERUN_POST_FUZZ_HOOKS_BY_TAG \
+        if rerun \
+        else _ALL_POST_FUZZ_HOOKS_BY_TAG
```

Comment:
I'm a little confused here - why are we limiting post_fuzz_hooks to only the rerun ones?

Reply:
`fuzz_lightyear/request.py` passes either `rerun=True` or `rerun=False` to this function depending on whether the request is being reused. If the request is being reused (i.e. `rerun=True`) then we only want to run the hooks that were marked as `rerun=True`, and so we should only get the rerun hooks.